src/EFCore.PG/Query/Expressions/Internal/PgFunctionExpression.cs

```diff
@@ -307,14 +303,44 @@ public override int GetHashCode()
             }
         }
 
-        /// <summary>
-        ///     Creates a <see cref="string" /> representation of the Expression.
-        /// </summary>
-        /// <returns>A <see cref="string" /> representation of the Expression.</returns>
+        /// <inheritdoc />
         public override string ToString()
-            => (Instance != null ? Instance + "." : Schema != null ? Schema + "." : "") +
-            $"{FunctionName}({string.Join("", "", PositionalArguments)}" +
-            (NamedArguments.Count > 0 ? ", " + (string.Join("", "", NamedArguments.Select(a => $"{a.Key} => {a.Value}"))) : "") +
-            ")";
+        {
+            StringBuilder sb = new StringBuilder();
```

Comment:
Let's revert this too, please. ToString is really not a perf-sensitive function, so converting a 3-line expression into this big alternative seems to be making things worse.

Reply:
I updated this to follow c1abc10deb487f5cab716c4e0d30b734208489a9, but it does seem error prone to maintain the code in separate locations... Will revert now.